### PR TITLE
Add public documentation for kubelet/apis/config

### PR DESCRIPTION
**Pull request by irbull**
The `kubelet/apis/config` package had golint failures. This change-set
removes the `kubelet/apis/config` package from the `.golint_failures`
and fixes the golint errors. In particular, this change-set adds
documentation to the public types in the package.

This change-set works towards issue #68026

**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespace from that line:
>
> /kind api-change
> /kind bug
> /kind cleanup
> /kind design
> /kind documentation
> /kind failing-test
> /kind feature
> /kind flake

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

### CODE REVIEW

File: pkg/kubelet/apis/config/types.go
Comment (mattjmcnaughton) [inline pkg/kubelet/apis/config/types.go]: Nit: I'm not sure the second sentence, where you list what the struct includes, is entirely necessary. I worry it has the ability to grow out of date, should fields be added/removed from the struct. More important, in my mind, is capturing the purpose of the struct, and I believe your first sentence does that successfully.

File: pkg/kubelet/apis/config/types.go
Comment (mattjmcnaughton) [inline pkg/kubelet/apis/config/types.go]: Same nit as above :)

File: pkg/kubelet/apis/config/types.go
Comment (mattjmcnaughton) [inline pkg/kubelet/apis/config/types.go]: Same nit as above :)

**Review by mattjmcnaughton — COMMENTED** [review]
Nice work here and welcome to k8s!!

A small nit from me, but otherwise this looks great. Ping me when you've had a chance to look at my comments and I can apply the next steps.

/release-note-none
/ok-to-test

File: pkg/kubelet/apis/config/types.go
Comment (irbull) [reply pkg/kubelet/apis/config/types.go]: Thanks. I was wondering about this when I wrote it. I'll fix this up. 

What is standard way to fix this, add another commit or `--amend` and force push back to this branch so we only have a single commit?

Comment (irbull) [reply pkg/kubelet/apis/config/types.go]: I have pushed a new commit to this PR that addresses these three comments. Should I "resolve" these conversations, or is that the responsibility of the original commenter?

Comment (mattjmcnaughton) [reply pkg/kubelet/apis/config/types.go]: Either as fine - at some point before this diff is merged, I'd recommend squashing your commits into one (although the merge-robot tool may actually do so automatically when it merges the PR... I'm not positive).

Typically, I use `--amend`, unless I think my reviewer will want to look at the pr commit by commit.

Re resolving the comments, feel free to go ahead and resolve them :)

**Review by mattjmcnaughton — APPROVED** [review]
/lgtm

File: pkg/kubelet/apis/config/types.go
Comment (irbull) [reply pkg/kubelet/apis/config/types.go]: Thanks, sorry for all the churn on this PR @mattjmcnaughton. I've squashed and rebased on `mater`.

**Review by mattjmcnaughton — APPROVED** [review]
/lgtm

Thank you for your work here :) Just a heads up, I think that its either KubeCon (or the week leading up to KubeCon), so getting approval from an `owner` may take slightly longer than usual :)

**Comment by mtaufen** [comment]
It might be better to include any additional documentation here on the external types, e.g. https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/kubelet/config/v1beta1/types.go instead. We haven't really documented the internal types, since they're just used, well, internally, and that way we only need to update docs in one place (the external types).

**Comment by irbull** [comment]
@mtaufen How do the types in `staging/src/k8s.io/kubelet/config/v1beta1/types.go` relate to those in `pkg/kubelet/apis/config/types.go`? Is one generated from the other, are the clones? 

I can also add the docs to `staging/src/k8s.io/kubelet/config/v1beta1/types.go` if it would help. But 
from what I can tell #68026 is about fixing the GoLint errors, and as configured, GoLint requires all Public Go APIs to contain code docs.

**Comment by mtaufen** [comment]
You can find more info about the relationship here: https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md

Configuration is written against the versioned, external types, and converted at runtime to the internal type either (a) for use by the runtime implementation or (b) for conversion back out to a different version of the external type schema. Both types are implemented by hand, the internal is a superset of the externals, and conversions, deep copies, etc. are typically, but not always, auto-generated.

I'm not opposed to fixing the failures, but worth noting that virtually all of the other Kubernetes-style API definitions (pkg/apis, cmd/kubeadm/apis, pkg/proxy/apis, staging/src/k8s.io/api, etc.) are opted out of golint, probably since the comments aren't following the linter conventions one way or another. It might be worth fixing them all at once.

**Comment by cblecker** [comment]
/approve
for `hack/`
/unassign

ping @mtaufen

**Pull request merged**
